src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +565,41 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
-      }
+  const loadWdsSchemaAndUrl = async () => {
+    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd && isAzureWorkspace) {
+      await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
+      })
+    }
+  }
+
+  const getWdsUrl = apps => {
```

Comment:
It looks like this function doesn't depend on anything from this component. Could it be moved out of this component to the top level of some module (maybe a data layer module)? Or, would it make sense for this to live in WdsDataTableProvider or an Ajax method that wraps `Apps.getV2AppInfo`?